File: 数组/设计有序流.java
package 数组;

import java.util.ArrayList;
import java.util.List;

public class 设计有序流 {
}

class OrderedStream {
    String[] stream;
    int ptr;
    public OrderedStream(int n) {
        stream = new String[n+1];
        ptr = 1;
    }
    public List<String> insert(int idKey, String value) {//类的方法
        List<String> res =new ArrayList<String>();
        stream[idKey] = value;
        while(ptr<stream.length && stream[ptr]!=null) {
            res.add(stream[ptr]);
            ptr++;
        }
        return res;
    }
}

File: 数组/设计有序流.go
package 数组

import "fmt"

type OrderedStream struct {
	stream []string
	ptr    int
}

func Constructor(n int) OrderedStream {
	stream := make([]string, n+1)
	ptr := 1
	return OrderedStream{stream, ptr}
}

func (this *OrderedStream) Insert(idKey int, value string) []string {
	var res []string
	this.stream[idKey] = value
	for this.ptr < len(this.stream) && this.stream[this.ptr] != "" {
		res = append(res, this.stream[this.ptr])
		this.ptr++
	}
	return res
}

func main() {
	fmt.Printf("fuck you")
}
